### PR TITLE
Fix broken page title/icon and archival banner

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR fixes setting page titles and icons. Previously this was broken as we were using the `updateEntity` mutation rather than the special cased page modification mutation.

This also fixes page archival banner that wasn't being loaded correctly.

## 🔍 What does this change?

- Adds hooks for updating page titles and icons
- Fixes usage of `updateEntity` for the page title
- Fixes page archival banner
- Adds missing `icon` property type for the page entity type
 
## 🛡 What tests cover this?

-

## ❓ How to test this?

1.  Checkout the branch, start external services and `yarn dev`
1. Open the workspace and sign in
1. Create a page and edit its title/icon

## 📹 Demo

image

### CODE REVIEW

File: packages/hash/frontend/src/components/hooks/useUpdatePageTitle.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    ],
+    [],
+  );
+
+  const updatePageTitle = useCallback(
+    async (value: string, ownedById: string, pageEntityId: string) => {
+      await updatePageFn({
+        variables: {
+          entityId: pageEntityId,
+          updatedProperties: { title: value },
```
Comment (benwerner01) [inline packages/hash/frontend/src/components/hooks/useUpdatePageTitle.ts:41]: Nit: argument naming
```suggestion
  const updatePageTitle = useCallback(
    async (title: string, ownedById: string, pageEntityId: string) => {
      await updatePageFn({
        variables: {
          entityId: pageEntityId,
          updatedProperties: { title },
```

File: packages/hash/frontend/src/components/hooks/useUpdatePageIcon.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    ],
+    [],
+  );
+
+  const updatePageIcon = useCallback(
+    async (value: string, ownedById: string, pageEntityId: string) => {
+      await updatePageFn({
+        variables: {
+          entityId: pageEntityId,
+          updatedProperties: { icon: value },
```
Comment (benwerner01) [inline packages/hash/frontend/src/components/hooks/useUpdatePageIcon.ts:41]: Nit: argument naming:
```suggestion
  const updatePageIcon = useCallback(
    async (icon: string, ownedById: string, pageEntityId: string) => {
      await updatePageFn({
        variables: {
          entityId: pageEntityId,
          updatedProperties: { icon },
```

File: packages/hash/frontend/src/components/PageIconButton.tsx [hunk 3]
```diff
@@ -43,8 +42,10 @@ export const PageIconButton = ({
     popupId: "emoji-picker",
   });
 
-  const { updateEntity, updateEntityLoading } =
-    useBlockProtocolUpdateEntity(true);
+  // const { updateEntity, updateEntityLoading } =
+  //   useBlockProtocolUpdateEntity(true);
+
+  const [updatePageIcon, { updatePageIconLoading }] = useUpdatePageIcon();
```
Comment (benwerner01) [inline packages/hash/frontend/src/components/PageIconButton.tsx:48]: Can we delete this?
```suggestion
  const [updatePageIcon, { updatePageIconLoading }] = useUpdatePageIcon();
```

**Review by benwerner01 — COMMENTED** [review]
Thanks Ahmad, looks good to me and is working locally. Have left a few nits and one thing that might be worth addressing

File: packages/hash/frontend/src/components/PageIconButton.tsx [hunk 4]
```diff
@@ -43,8 +42,10 @@ export const PageIconButton = ({
     popupId: "emoji-picker",
   });
 
-  const { updateEntity, updateEntityLoading } =
-    useBlockProtocolUpdateEntity(true);
+  // const { updateEntity, updateEntityLoading } =
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/hash/frontend/src/components/PageIconButton.tsx:45]: Nope, will remove

**Review by benwerner01 — APPROVED** [review]

**Pull request merged**